src/test/scala/special/sigma/SigmaDslTest.scala

```diff
@@ -973,4 +973,16 @@ class SigmaDslTest extends PropSpec
     forAll { x: (Boolean, Boolean) => eq(x) }
   }
 
+  property("lazy || boolean equivalence") {
+    checkEq(func[Boolean, Boolean]("{ (x: Boolean) => x || (1/0 == 1) }")){ (x: Boolean) =>
+      x || (1 / 0 == 1)
+    }(true)
+  }
+
+  property("lazy && boolean equivalence") {
+    checkEq(func[Boolean, Boolean]("{ (x: Boolean) => (x && (1/0 == 1)) == false }")){ (x: Boolean) =>
+      (x && (1 / 0 == 1)) == false
+    }(false)
+  }
```

Comment:
@greenhat Great! Let's add more eq tests here with 2, 3, 4 nested && and || levels in both left and right arguments.
Around 15 tests that cover all possible situations.

Reply:
Done.